lib/compress/zstd_compress.c

```diff
@@ -1959,6 +1960,12 @@ static size_t ZSTD_resetCCtx_internal(ZSTD_CCtx* zc,
         zc->outBuffSize = buffOutSize;
         zc->outBuff = (char*)ZSTD_cwksp_reserve_buffer(ws, buffOutSize);
 
+        /* block splitter ctx */
+        if (ZSTD_CParams_useBlockSplitter(&params->cParams)) {
+            /* Silence -Wcast-align with cast to void* */
+            zc->blockSplitCtx = (ZSTD_blockSplitCtx*)(void*)ZSTD_cwksp_reserve_buffer(ws, sizeof(ZSTD_blockSplitCtx));
```

Comment:
`-Wcast-align` has a point here. You'll need correct alignment here or certain platforms will fail.

This probably needs to go in the `ZSTD_cwksp_reserve_object()` space, or if that doesn't work just put it directly into the CCtx.

Reply:
Yeah you're right. I think this will just rest directly in the CCtx. If I want to save space when we're not using block splitting, a wksp object doesn't seem to work properly with the different cctx reuse cases, since objects are assumed to remain the same throughout compressions.

If I wanted to make it work somehow, it would have to be the same as the other objects in that they stay the same throughout different cctx re-uses. And if we have to do that anyways, we may as well just keep it in the cctx.

Reply:
> Though, this does mean we will just have to eat the >1K additional heap cost.

That should be fine, thats much less than the total CCtx allocation. E.g. we have 500KB space for sequences, which is way overkill in most cases. So there is lower hanging fruit.